### ISSUE TITLE
Set priorities when events overlap

### ISSUE THREAD

**Issue by aron1127** [zjfjack/JZCalendarWeekView#72]
(I do not speak English well. If you do not understand, please reply.)

Hi. Thank you very much for creating a useful library before asking questions.

If there are several events on the same date and time,
I want to show only high priority events.

I am in the `JZWeekViewFlowLayout.swift` file

`adjustItemsForOverlap (sectionItemAttributes, inSection: section, sectionMinX: sectionX, currentSectionZ: zIndexForElementKind (JZSupplementaryViewKinds.eventCell))`

By disabling the code, able to solve it roughly.

But the event I want does not appear at the top. (It is displayed randomly.)
Is there a way to prioritize or show the event i want first?

And I have a simple question.
**Limited date range: start time and end Time (vertical) in CalendarView**
I really want it. When will it be implemented?

**Comment by zjfjack** [comment]
Actually, I have one question regarding this requirement because a few people mentioned this feature in this project.

Do you want to hide all the other dates not in this range, and you cannot see those time or just simply don't allow users to scroll out of this range but they can still see those time.

I am not sure whether you can understand my meaning. If you cannot, could you send me a picture for your requirement? Thanks

**Comment by RanjeetStiga** [comment]
> Hi. Thank you very much for creating a useful library before asking questions.
> 
> I have a simple question.
> **Limited date range: start time and end Time (vertical) in CalendarView**
> I really want it. When will it be implemented?

Have you achieve this ?

**Comment by zjfjack** [comment]
@RanjeetStiga #76 Your case is cannot see those time?

Could you send some pictures here.

**Comment by aron1127** [comment]
It does not matter to me how it works.

If I set the time range from 08:00 to 22:00,  should not show before 8:00 or after 22:00. 

I also want to have the ability to update the time settings like forceReload.

Finally, if I set it after 10 o'clock and add an event before 10 o'clock, it should not cause an error. (exception handling)
This is because there are a number of cases in the api of my app server.

If you're still not sure, I'll provide a picture of our app.

**Comment by zjfjack** [comment]
Is it similar to Apple Calendar event details page? @aron1127 @RanjeetStiga @nttra83 @inlovewithswift

Users are not allowed to scroll both horizontally and vertically and view just displays the RowHeader (Time) without ColumnHeader (Date). That's how Apple Calendar does.

If you guys have the same requirements, I can try to unify them and do something for it.

Can we have a discussion here and ignore the #66 and #76 issues?

I really need you guys send me some of your requirements picture, then it should be easier for the discussion and implementation.

Thanks

**Comment by PMA22** [comment]
Hi. A need something like this too. For example I have first event from 8:00 to 8:30 and last event from 15:00 to 17:00. I want day calendar from 8:00 to 17:00. I know how many hours it is (9). I calculate height (my calendar is without scroll) by hour which I want display and row height (9 * hourHeight). This height I add to customCalendarView. For here all is ok.
My problem is which part of collection view with events is displayed. I have part from currentTimeline,  not from time which I need. In this case I want calendar from 8:00 to 17:00 with currentTimeline at 10:00, but I have calendar from 10:00 to 19:00 with currentTimeline on top. 
Is there some way how resolve this?

**Comment by aron1127** [comment]
* I do not speak English well, so word choice may not be appropriate.

I did not want a new view like the Apple Calendar event details page.
My requirements are so simple that I do not need a picture.

The current CalendarWeekView is displayed from 0:00 to 24:00 unconditionally.
I just wanted to set this up.

If i set the time from 09:00 to 20:00, there should be no view before 9:00 or after 20:00. (Do not scroll too)

In addition, when the short interval is set from 08:00 to 10:00
![asdf](https://user-images.githubusercontent.com/28154360/58141458-727c8f00-7c7e-11e9-8db3-90f6919e0a96.png)

I'd want to show it like this image.

**Comment by RanjeetStiga** [comment]
any Update ?

**Comment by zjfjack** [comment]
@RanjeetStiga I was working on the redesign of pagination which is a major release for the whole project. I released the latest 0.7.0 version yesterday. 

For your requirements, you guys should have a discussion here with your **design pictures** and make a conclusion.

Otherwise, I cannot achieve all your requirements. You can put your requirements and pictures first.

Thanks

**Comment by aron1127** [comment]
Hi. When will it be updated?
I currently need this very much.

Perhaps my design is a requirement for everyone.
because the requirement is mentioned and there is no other design until now.

**Comment by zjfjack** [comment]
Hi @aron1127,

I haven't started it yet. I am still waiting for others' requirements.
Regarding your requirements, I understand what you need vertically. As for the horizontal range, do you want them to have all the same vertical range as well? It means you can only scroll vertically from the start time to the end time with a vertical range even if you horizontally scroll to the other page.

Am I right? Do you guys also have similar requirements to his? Do you agree with this solution?
@PMA22 @nttra83 @inlovewithswift

**Comment by aron1127** [comment]
That's right. Even if I horizontally scroll to the other page, the vertical range should be the same as the setting.

I hope others will be able to comment quickly ... :(

**Comment by RanjeetStiga** [comment]
Hi
When will be update?

**Comment by aron1127** [comment]
Hi @zjfjack 
Do you have any good news about this issue?
I am waiting for updates..
Could you tell me when it will be updated?

**Comment by barb94** [comment]
Hi @zjfjack 
It would be very useful for me if you can implement this feature.
I have same requirements as @aron1127 . 
In example: i need to show events that are always in the range 8am-8pm (working time ∓ 1h) so even if i scroll to the next horizontal range, the vertical hours range must be the same. This will help my users to avoid to always manually scroll to 8am, because surely they won't have any 12pm-8am event.
Thank you for your time, hope to hear you soon about this.

**Comment by aron1127** [comment]
Hello @zjfjack
I have been delayed in development and have to re-plan.
I want you to let me know when the development will be completed.
Plz answer the your development schedule.
I'm sorry I kept asking questions...

**Comment by zjfjack** [comment]
Hi @aron1127 and @barb94,

I was quite busy on my work for the last two months and I am not sure when I can start implementing this. Let me try to do it this weekend. If I can achieve it easily, then I will make the pull request. If I cannot, then I don't think I can finish them recently.

Sorry about that.

**Comment by aron1127** [comment]
Thank you for your answer.
I hope it will be completed by August.
I look forward to the good results!!

**Comment by zjfjack** [comment]
Hi @aron1127 @barb94,

I have some good news to you guys, which is that I am currently working on this feature. I've done some basic layout stuff.

There are still some issues for longPress and reload but it is showing! Hopefully, I won't have some complicated problems when resolving the left issues and release it as soon as possible.

Let me show you a screenshot of this new feature to make sure this is what you want. 

Thanks!

![Simulator Screen Shot - iPhone 8 - 2019-09-01 at 22 55 47](https://user-images.githubusercontent.com/4811838/64076771-4906e100-cd0c-11e9-93d6-73485fc6b4ee.png)

**Comment by RanjeetStiga** [comment]
> @RanjeetStiga #76 Your case is cannot see those time?
> 
> Could you send some pictures here.
Hi Same issue  #76 @aron1127 .

**Comment by barb94** [comment]
Hello @zjfjack ,
yes, this is exactly what I need!

Thank your for your work

**Comment by barb94** [comment]
Hi @zjfjack , 
is there any update about this feature?
Thanks!

**Comment by dnhrtmn** [comment]
Hi @zjfjack,

was this function implemented in the meantime?

Thank you in advance!

Regards

**Comment by PoojaSoftRadix** [comment]
Hi @zjfjack 
is there any update about this feature?
Thanks!
